src/main/scala/viper/silver/plugin/standard/termination/transformation/ExpTransformer.scala

```diff
@@ -209,6 +262,9 @@ trait ExpTransformer extends ErrorReporter {
       If(Not(c)(c.pos), els, EmptyStmt)(i.pos, i.info, i.errT)
     case i@If(c1, Seqn(Seq(If(c2, thn, EmptyStmt)), Nil), EmptyStmt) => // combine nested if clauses
       If(And(c1, c2)(), thn, EmptyStmt)(i.pos, i.info, i.errT)
+    case If(TrueLit(), thn, _) => thn // remove trivial If conditions (in particular effective together with the next 2 simplifications)
```

Comment:
Not really related to this PR, but we could move this code to Silver's existing Simplifier.